File: sdks/python/src/echo_agent_sdk/command_cell_values.py
from __future__ import annotations

from enum import Enum


class CommandCellPhase(str, Enum):
    PREPARED = "prepared"
    QUEUED = "queued"
    RUNNING = "running"
    SUCCEEDED = "succeeded"
    FAILED = "failed"
    CANCELLED = "cancelled"
    LAUNCH_FAILED = "launch_failed"

    def as_str(self) -> str:
        return self.value

    def is_terminal(self) -> bool:
        return self in {
            self.SUCCEEDED,
            self.FAILED,
            self.CANCELLED,
            self.LAUNCH_FAILED,
        }

File: sdks/typescript/src/command_cell_values.ts
/** Command-cell renderer phases projected without owning execution. */
export const CommandCellPhase = Object.freeze({
  Prepared: "prepared",
  Queued: "queued",
  Running: "running",
  Succeeded: "succeeded",
  Failed: "failed",
  Cancelled: "cancelled",
  LaunchFailed: "launch_failed",
} as const);

export type CommandCellPhase = (typeof CommandCellPhase)[keyof typeof CommandCellPhase];

export function commandCellPhaseAsStr(phase: CommandCellPhase): string {
  if (!Object.values(CommandCellPhase).includes(phase)) {
    throw new TypeError("invalid command cell phase");
  }
  return phase;
}

export function commandCellPhaseIsTerminal(phase: CommandCellPhase): boolean {
  return phase === CommandCellPhase.Succeeded
    || phase === CommandCellPhase.Failed
    || phase === CommandCellPhase.Cancelled
    || phase === CommandCellPhase.LaunchFailed;
}
